include/usearch/usearch.hpp

```diff
@@ -7,26 +7,39 @@
  *  @copyright Copyright (c) 2023
  */
 #pragma once
-#include <algorithm> // `std::sort_heap`
-#include <atomic>    // `std::atomic`
-#include <bitset>    // `std::bitset`
-#include <climits>   // `CHAR_BIT`
-#include <cmath>     // `std::sqrt`
-#include <cstring>   // `std::memset`
-#include <mutex>     // `std::unique_lock` - replacement candidate
-#include <random>    // `std::default_random_engine` - replacement candidate
-#include <unistd.h>  // `open`, `close`
-#include <utility>   // `std::exchange`
-#include <vector>    // `std::vector`
 
 #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#define NOMINMAX // define this macro to prevent the definition of min/max macros in Windows.h
```

Comment:
This will affect compiling every file where `usearch.hpp` is included. There is a way without defining it. We may want to provide our own `min` and `max` instead.

Reply:
We define our own `min` and `max` functions, which is why this macro is required. There is also a conflict regarding that in Clipp.

Reply:
Where do we define the `min` and the `max`? The `clipp` isn't a problem, cause we only include it in the benchmark executable, while this is a library header, that others will pull.

Reply:
Oh, I see. That function is never used, and I removed it. Please fetch the newest version from `main-dev`.